### PR TITLE
Cache AttachmentProvider

### PR DESCRIPTION
**Pull request by felixbarny**

### CODE REVIEW

File: apm-agent-attach/src/main/java/co/elastic/apm/attach/CachedAttachmentProvider.java [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+ */
+package co.elastic.apm.attach;
+
+import net.bytebuddy.agent.ByteBuddyAgent;
+
+/**
+ * Successive attachments would lead to the VirtualMachine class to be loaded from multiple class loaders
+ * (see net.bytebuddy.agent.ByteBuddyAgent.AttachmentProvider.ForStandardToolsJarVm#attempt()).
+ * That leads to a UnsatisfiedLinkError on Java 7 and 9 because the native library libattach can only be loaded by one class loader.
+ * By caching the {@link net.bytebuddy.agent.ByteBuddyAgent.AttachmentProvider.Accessor}, the same
```
Comment (eyalkoren) [inline apm-agent-attach/src/main/java/co/elastic/apm/attach/CachedAttachmentProvider.java]: Your sentence ends unexpectedly 🙂

File: apm-agent-attach/src/main/java/co/elastic/apm/attach/CachedAttachmentProvider.java [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+ */
+public class CachedAttachmentProvider implements ByteBuddyAgent.AttachmentProvider {
+    private final Accessor accessor;
+
+    public CachedAttachmentProvider(ByteBuddyAgent.AttachmentProvider provider) {
+        this.accessor = provider.attempt();
```
Comment (eyalkoren) [inline apm-agent-attach/src/main/java/co/elastic/apm/attach/CachedAttachmentProvider.java]: This means only one AttachementProvider can be attempted. Is that the case always?

Comment (felixbarny) [reply apm-agent-attach/src/main/java/co/elastic/apm/attach/CachedAttachmentProvider.java]: There is a `AttachmentProvider.Compound` which tries multiple ones and returns the first successful attempt.

File: apm-agent-attach/src/main/java/co/elastic/apm/attach/CachedAttachmentProvider.java (hunk 1 above)
Comment (felixbarny) [reply apm-agent-attach/src/main/java/co/elastic/apm/attach/CachedAttachmentProvider.java]: 😱

**Pull request merged**